Top/main.c

```diff
@@ -242,405 +237,403 @@ int32_t csoundCompileArgs(CSOUND *csound, int32_t argc, const char **argv)
+    csoundNotifyFileOpened(csound, csound->xfilename, CSFTYPE_EXTRACT_PARMS, 0,
+                           0);
+    if (O->msglevel || O->odebug)
+      csound->Message(csound, Str("  ... extracting ...\n"));
+    scxtract(csound, csound->scstr, xfile);
+    fclose(xfile);
+    csound->tempStatus &= ~csPlayScoMask;
+  }
+  if (O->msglevel || O->odebug)
+    csound->Message(csound, Str("\t... done\n"));
+  /* copy sorted score name */
+  csound->playscore = csound->scstr;
+  /* IV - Jan 28 2005 */
+  print_benchmark_info(csound, Str("end of score sort"));
+  if (O->syntaxCheckOnly) {
+    csound->Message(csound, Str("Syntax check completed.\n"));
+    // return CSOUND_EXITJMP_SUCCESS;
+    if (compiledOk)
+      return CSOUND_EXITJMP_SUCCESS;
+    return CSOUND_ERROR;
+  }
+  return CSOUND_SUCCESS;
 }
 
-extern int32_t  playopen_dummy(CSOUND *, const csRtAudioParams *parm);
+extern int32_t playopen_dummy(CSOUND *, const csRtAudioParams *parm);
 extern void rtplay_dummy(CSOUND *, const MYFLT *outBuf, int32_t nbytes);
-extern int32_t  recopen_dummy(CSOUND *, const csRtAudioParams *parm);
-extern int32_t  rtrecord_dummy(CSOUND *, MYFLT *inBuf, int32_t nbytes);
+extern int32_t recopen_dummy(CSOUND *, const csRtAudioParams *parm);
+extern int32_t rtrecord_dummy(CSOUND *, MYFLT *inBuf, int32_t nbytes);
 extern void rtclose_dummy(CSOUND *);
-extern int32_t  audio_dev_list_dummy(CSOUND *, CS_AUDIODEVICE *, int32_t);
-extern int32_t  midi_dev_list_dummy(CSOUND *csound, CS_MIDIDEVICE *list, int32_t isOutput);
+extern int32_t audio_dev_list_dummy(CSOUND *, CS_AUDIODEVICE *, int32_t);
+extern int32_t midi_dev_list_dummy(CSOUND *csound, CS_MIDIDEVICE *list,
+                                   int32_t isOutput);
 extern int32_t DummyMidiInOpen(CSOUND *csound, void **userData,
-                           const char *devName);
-extern int32_t DummyMidiRead(CSOUND *csound, void *userData,
-                         unsigned char *buf, int32_t nbytes);
+                               const char *devName);
+extern int32_t DummyMidiRead(CSOUND *csound, void *userData, unsigned char *buf,
+                             int32_t nbytes);
 extern int32_t DummyMidiOutOpen(CSOUND *csound, void **userData,
-                     const char *devName);
+                                const char *devName);
 extern int32_t DummyMidiWrite(CSOUND *csound, void *userData,
-                   const unsigned char *buf, int32_t nbytes);
-
+                              const unsigned char *buf, int32_t nbytes);
 
 PUBLIC int32_t csoundStart(CSOUND *csound) // DEBUG
 {
-    OPARMS  *O = csound->oparms;
-    int32_t     n;
+  OPARMS *O = csound->oparms;
+  int32_t n;
 
-    /* if a CSD was not used, check options */
-    if (csound->csdname == NULL)
-          checkOptions(csound);
+  /* if a CSD was not used or options were not checked, check options */
+  if (csound->csdname == NULL && !csound->options_checked) {
+    checkOptions(csound);
```

Comment:
Should this set `csound->options_checked = 1` here?

Reply:
This was marked resolved but I didn't see a code change or a reply, unresolving for now.

Reply:
Maybe my comment did not come out for some reason.

It's not relevant as csoundSetOptions only runs before first compilation, never after. This code here can only be reached again after reset.